timemachine/fe/rbfe.py

```diff
@@ -354,122 +305,43 @@ def estimate_free_energy_given_initial_states(initial_states, protocol, temperat
-        bound_impls = None
-        cur_batch_U_fns = None
-        cur_frames, cur_boxes = sample(initial_state, protocol)
-        bound_impls = [p.bound_impl(np.float32) for p in initial_state.potentials]
-        cur_batch_U_fns = get_batch_U_fns(bound_impls)
-
-        if lamb_idx in keep_idxs:
-            stored_frames.append(cur_frames)
-            stored_boxes.append(cur_boxes)
-
-        if lamb_idx > 0:
-
-            ukln_by_component = []
-
-            # loop over bond, angle, torsion, nonbonded terms etc.
-            for u_idx, (prev_U_fn, cur_U_fn) in enumerate(zip(prev_batch_U_fns, cur_batch_U_fns)):
-                u_00 = beta * prev_U_fn(prev_frames, prev_boxes)
-                u_01 = beta * prev_U_fn(cur_frames, cur_boxes)
-                u_10 = beta * cur_U_fn(prev_frames, prev_boxes)
-                u_11 = beta * cur_U_fn(cur_frames, cur_boxes)
-                ukln_by_component.append([[u_00, u_01], [u_10, u_11]])
-
-                fwd_delta_u = u_10 - u_00
-                rev_delta_u = u_01 - u_11
-                plot_axis = all_axes[lamb_idx - 1][u_idx]
-                plot_work(fwd_delta_u, rev_delta_u, plot_axis)
-                plot_axis.set_title(U_names[u_idx])
-
-            # sanity check - I don't think the dG calculation commutes with its components, so we have to re-estimate
-            # the dG from the sum of the delta_us as opposed to simply summing the component dGs
-
-            # (energy components, energy fxns = 2, sampled states = 2, frames)
-            ukln_by_component = np.array(ukln_by_component, dtype=np.float64)
-            total_fwd_delta_us = (ukln_by_component[:, 1, 0, :] - ukln_by_component[:, 0, 0, :]).sum(axis=0)
-            total_rev_delta_us = (ukln_by_component[:, 0, 1, :] - ukln_by_component[:, 1, 1, :]).sum(axis=0)
-            total_df, total_df_err = bar_with_bootstrapped_uncertainty(total_fwd_delta_us, total_rev_delta_us)
-
-            plot_axis = all_axes[lamb_idx - 1][u_idx + 1]
-
-            plot_BAR(
-                total_df,
-                total_df_err,
-                total_fwd_delta_us,
-                total_rev_delta_us,
-                f"{prefix}_{lamb_idx-1}_to_{lamb_idx}",
-                plot_axis,
-            )
-
-            total_dG = total_df / beta
-            total_dG_err = total_df_err / beta
+    # run n_lambdas simulations in sequence
+    u_kln_by_component_by_lambda, stored_frames, stored_boxes = run_sequential_sims_given_initial_states(
+        initial_states, protocol, temperature, keep_idxs
+    )
 
-            all_dGs.append(total_dG)
-            all_errs.append(total_dG_err)
-            ukln_by_component_by_lambda.append(ukln_by_component)
+    # "pair BAR" free energy analysis
+    kBT = BOLTZ * temperature
+    beta = 1 / kBT
 
-            print(
-                f"{prefix} BAR: lambda {lamb_idx-1} -> {lamb_idx} dG: {total_dG:.3f} +- {total_dG_err:.3f} kJ/mol",
-                flush=True,
-            )
+    all_dGs = []
```

Comment:
Seems to duplicate code in make_overlap_detail_figure, should this be made into a func in bar?

Reply:
Yes -- see also suggestion https://github.com/proteneer/timemachine/pull/936#discussion_r1093482189

Reply:
Duplication resolved by applying suggestion in https://github.com/proteneer/timemachine/pull/936#discussion_r1093482189